IOS debug build fails non deterministically on local environment

Right after upgrading react-native to 0.67.5 we have an IOS build failure with the following error message : 

```log
** BUILD FAILED **


The following build commands failed:
	CompileC /Users/siddarthkumar/Library/Developer/Xcode/DerivedData/
	StatusIm-fxiitjbbybrcsqcxqartjfyvqxdo/Build/Intermediates.noindex/Pods.build/
	Debug-iphonesimulator/FBReactNativeSpec.build/Objects-normal/x86_64/
	FBReactNativeSpec-generated-fd168fa4da568d55fc39af3f2cac5990.o 
	/Users/siddarthkumar/code/experiments/status-mobile/node_modules/
	react-native/React/FBReactNativeSpec/FBReactNativeSpec/out/FBReactNativeSpec-generated.mm 
	normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler 
	(in target 'FBReactNativeSpec' from project 'Pods')
	
	CompileC /Users/siddarthkumar/Library/Developer/Xcode/DerivedData/
	StatusIm-fxiitjbbybrcsqcxqartjfyvqxdo/Build/Intermediates.noindex/Pods.build/
	Debug-iphonesimulator/FBReactNativeSpec.build/Objects-normal/arm64/
	FBReactNativeSpec-generated-fd168fa4da568d55fc39af3f2cac5990.o 
	/Users/siddarthkumar/code/experiments/status-mobile/node_modules/
	react-native/React/FBReactNativeSpec/FBReactNativeSpec/out/FBReactNativeSpec-generated.mm 
	normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler 
	(in target 'FBReactNativeSpec' from project 'Pods')
	
(2 failures)
```

Have logged this issue to track any more such build failures and to investigate and find their fixes.
However we do not see such build failures on CI.

--- a/ios/StatusIm.xcodeproj/project.pbxproj
+++ b/ios/StatusIm.xcodeproj/project.pbxproj
@@ -544,7 +544,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = "/usr/bin/env sh";
-			shellScript = "set -o errexit\nexport NODE_BINARY=\"${NODE_BINARY:-node}\"\nexport NODE_ARGS=\" --openssl-legacy-provider --max-old-space-size=16384 \"\n\nbash -x ../node_modules/react-native/scripts/react-native-xcode.sh > ./logs/react-native-xcode.log 2>&1";
+			shellScript = "set -o errexit\nexport NODE_BINARY=\"${NODE_BINARY:-node}\"\nexport NODE_ARGS=\"${NODE_ARGS:- --max-old-space-size=16384 }\"\n\nbash -x ../node_modules/react-native/scripts/react-native-xcode.sh > ./logs/react-native-xcode.log 2>&1";
 		};
 		3AAD2AD524A3A60E0075D594 /* Bundle React Native code and images */ = {
 			isa = PBXShellScriptBuildPhase;
@@ -558,7 +558,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = "/usr/bin/env sh";
-			shellScript = "set -o errexit\nexport NODE_BINARY=\"${NODE_BINARY:-node}\"\nexport NODE_ARGS=\" --openssl-legacy-provider --max-old-space-size=16384 \"\n\nbash -x ../node_modules/react-native/scripts/react-native-xcode.sh > ./logs/react-native-xcode.log 2>&1";
+			shellScript = "set -o errexit\nexport NODE_BINARY=\"${NODE_BINARY:-node}\"\nexport NODE_ARGS=\"${NODE_ARGS:- --max-old-space-size=16384 }\"\n\nbash -x ../node_modules/react-native/scripts/react-native-xcode.sh > ./logs/react-native-xcode.log 2>&1";
 		};
 		3AAD2AD724A3A60E0075D594 /* Run Script */ = {
 			isa = PBXShellScriptBuildPhase;

--- a/nix/mobile/ios/shells/nodejs.nix
+++ b/nix/mobile/ios/shells/nodejs.nix
@@ -5,6 +5,8 @@ mkShell {
   shellHook = ''
     # Fix for ERR_OSSL_EVP_UNSUPPORTED error.
     export NODE_OPTIONS="--openssl-legacy-provider";
+    # Same fix but for Xcode React Native script.
+    export NODE_ARGS="--openssl-legacy-provider --max-old-space-size=16384";
 
     # Fix Xcode using system Node.js version.
     export NODE_BINARY="${nodejs}/bin/node";

